design-docs/dependency-management-projectpriority.md

```diff
@@ -0,0 +1,49 @@
+# Use cases
+
+Gradle normally handles version conflicts by picking the highest version of the module.
+
+In rare cases this may result in a project dependency being replaced by a module dependency, which
+may be counter-intuitive.
+
+Other use cases depend on no priority being given to project dependencies.
+
+## Use case 1 (project priority required)
+
+The actual use case involved a a multi-repo system, where developers could choose a subset
+of projects to work on, but here is a simple case:
+
+We have 3 projects: `common`, `moduleA`, `moduleB`
+
+* `moduleA` has a dependency on `common`
+* `moduleB` depends on both `common` and `moduleA`
+* `settings.gradle` only includes `common` and `moduleB`. `moduleA` is being pulled from a repository.
+
+What happens:
```

Comment:
This is what happens currently in Gradle `master`. Please add what should happen with your change included - i.e. having called `resolutionStrategy.preferProjectModules`.